lib/kafka/sasl_scram_authenticator.rb

```diff
@@ -0,0 +1,171 @@
+require 'securerandom'
+require 'base64'
+
+module Kafka
+  SCRAM_SHA256 = 'SHA-256'.freeze
+  SCRAM_SHA512 = 'SHA-512'.freeze
+  class SaslScramAuthenticator
+    def initialize(username, password, mechanism: SCRAM_SHA256, logger: nil, connection:)
+      @username = username
+      @password = password
+      @mechanism = mechanism
+      @logger = logger
+      @connection = connection
+    end
+
+    def authenticate!
+      response = @connection.send_request(Kafka::Protocol::SaslHandshakeRequest.new('SCRAM-' + @mechanism))
+
+      unless response.error_code == 0 && response.enabled_mechanisms.include?('SCRAM-' + @mechanism)
+        raise Kafka::SaslScramError, "SCRAM-#{@mechanism} is not supported."
+      end
+
+      log_debug "authenticating #{@username} with scram, mechanism: #{@mechanism}"
+
+      @encoder = @connection.encoder
+      @decoder = @connection.decoder
+
+      begin
+        msg = first_message
+        log_debug "[scram] Sending client's first message: #{msg}"
```

Comment:
There's currently no precedent for "tagging" log lines with e.g. `[scram]`. Could you instead do `Sending first SCRAM client message`? The same for the rest of the log lines.

Reply:
Yes